modin/pandas/test/dataframe/test_binary.py

```diff
@@ -197,3 +197,51 @@ def test_equals():
         df_equals(modin_df3, modin_df2)
 
     assert modin_df1.equals(modin_df2._query_compiler.to_pandas())
+
+
+@pytest.mark.parametrize("is_test_series", [True, False], ids=["op_ser", "op_df"])
+@pytest.mark.parametrize(
+    "is_aligned_idx", [True, False], ids=["idx_aligned", "idx_not_aligned"]
+)
+def test_mismatched_row_partitions(is_aligned_idx, is_test_series):
+    data = [0, 1, 2, 3, 4, 5]
+    modin_df_one_part, pandas_df_one_part = create_test_dfs({"a": data, "b": data})
+    modin_df, pandas_df = modin_df_one_part.loc[:2], pandas_df_one_part.loc[:2]
+
+    modin_df_two_parts = modin_df.append(modin_df)
+    pandas_df_two_parts = pandas_df.append(pandas_df)
+
+    if is_aligned_idx:
+        modin_df_two_parts.index = pandas_df_two_parts.index = pandas_df_one_part.index
+
+    if is_test_series:
+        modin_res = modin_df_two_parts.a / modin_df_one_part.a
+        pandas_res = pandas_df_two_parts.a / pandas_df_one_part.a
+    else:
+        modin_res = modin_df_two_parts / modin_df_one_part
+        pandas_res = pandas_df_two_parts / pandas_df_one_part
+    df_equals(modin_res, pandas_res)
+
+
+@pytest.mark.parametrize(
+    "is_aligned_cols", [True, False], ids=["cols_aligned", "cols_not_aligned"]
+)
+def test_mismatched_col_partitions(is_aligned_cols):
```

Comment:
This test should work on current master because our binary operations are aligned on index.

Reply:
Hm, yes, I agree with you. I think we can delete this test.